### ISSUE TITLE
Azure ressource tagging

### ISSUE THREAD

**Issue by karstenmueller** [pendrica/chef-provisioning-azurerm#23]
How to simplify the tagging of all ressources? How to apply a tag set to all ressources?
Perhaps using a  parameter  for azure_resource_template will be a good way?

```ruby
azure_resource_template 'example' do
  parameters tagSet: '{ "department": "mydepartment", "environment": "test", "owner": "justme" }',
end
```

Then using
```
"tags": "[parameters('tagSet')]"
```
in the arm template.

What do you think?

**Comment by stuartpreston** [comment]
I haven't done any investigation on this but your way looks reasonable, in fact probably the only reasonable way as there's no such thing as tags on a resource template deploment.  Are you saying that this suggested approach does not work?

One problem I could see may be that the tagSet parameter possibly needs to be an Object type rather than a string.  See https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags for an example where the tags are defaulted in the parameter definition in the ARM template.  

Let me know how you get on and if you need help I'll try and spend some time recreating a better example.

**Comment by karstenmueller** [comment]
Yes, an object type is needed. Thanks for the hint!

Processing an object for the tag names and values is not supported so I had to specify the tag names in the template. See [azuredeploy.txt](https://github.com/pendrica/chef-provisioning-azurerm/files/609451/azuredeploy.txt) which is pretty much your example template with tags added.

I tried to specify a parameter for azure_resource_template like
```
tagvalues: '{ "environment": "test", "department": "mydep", "project": "chef-tagvalues", "owner": "justme" }'
```
but got an error: {"error":{"code":"InvalidTemplate","message":"Deployment template
 validation failed: 'The provided value for the template parameter 'tagvalues' at line '1' and colum
n '975' is not valid.'."}}

It would be nice if specifying the tag names in the template could be avoided.
